scout/adapter/mongo/omics_variant.py

```diff
@@ -0,0 +1,145 @@
+import logging
+from typing import Dict, Optional
+
+from scout.constants import OMICS_FILE_TYPE_MAP
+from scout.models.omics_variant import OmicsVariantLoader
+from scout.parse.omics_variant import parse_omics_file
+
+LOG = logging.getLogger(__name__)
+
+
+class OmicsVariantHandler:
+    def delete_omics_variants(self, case_id: str, file_type: str):
+        """Delete OMICS variants for a case"""
+        omics_file_type = OMICS_FILE_TYPE_MAP.get(file_type)
+        category = omics_file_type["category"]
+        sub_category = omics_file_type["sub_category"]
+        variant_type = omics_file_type["variant_type"]
+
+        LOG.info(
+            "Deleting old %s %s %s OMICS variants.",
+            variant_type,
+            sub_category,
+            category,
+        )
+
+        query = {
+            "case_id": case_id,
+            "variant_type": variant_type,
+            "category": category,
+            "sub_category": sub_category,
+        }
+        result = self.omics_variant_collection.delete_many(query)
+
+        LOG.info("%s variants deleted", result.deleted_count)
+
+    def set_samples(self, case_obj: dict, omics_model: dict):
+        """Internal member function to connect individuals.
+        OMICS variants do not have a genotype as such."""
+
+        samples = []
+
+        for ind in case_obj.get("individuals"):
+            if omics_model["sampleID"] == ind.get("individual_id"):
+                sample = {
+                    "sample_id": ind["individual_id"],
+                    "display_name": ind["display_name"],
+                    "genotype_call": "./1",
```

Comment:
Yes, it doesn't make biological sense. Neither does some of the calls we have for rather unspecific other variants really. And it makes it so much easier to reuse all downstream code instead of having to add in exceptions for these everywhere.